systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -545,6 +548,29 @@ private void checkPings(int messagesCount, Job job) {
         assertTrue("The consumer didn't consume any messages (no records_consumed message)", consumerSuccess);
     }
 
+    /**
+     * Greps logs from a pod which ran kafka-verifiable-consumer.sh
+     */
+    private void checkRecordsForConsumer(int messagesCount, Job job) {
+        String podName = jobPodName(job);
+        String log = podLog(podName);
+        Pattern p = Pattern.compile("^\\{.*\\}$", Pattern.MULTILINE);
+        Matcher m = p.matcher(log);
+        boolean consumerSuccess = false;
+        while (m.find()) {
+            String json = m.group();
+            String name2 = getValueFromJson(json, "$.name");
+            if ("records_consumed".equals(name2)) {
+                assertEquals(String.valueOf(messagesCount), getValueFromJson(json, "$.count"));
+                consumerSuccess = true;
+            }
+        }
+        if (!consumerSuccess) {
+            LOGGER.info("log from pod {}:\n----\n{}\n----", podName, indent(log));
+        }
+        assertTrue("The consumer didn't consume any messages (no records_consumed message)", consumerSuccess);
```

Comment:
The default value for `consumerSuccess` is `false` and when method finds `records_consumed` with the expected value for `$.count`, this method updates this value to `true`. Then, we match this value in JUnit method assertTrue(String message, boolean condition) and if a method argument `condition` is true, JUnit just marks this test as passed. Otherwise, if method argument `condition` is false, it means that method didn't find records and method need to print message `"The consumer didn't consume any messages (no records_consumed message)"` and fails this assertion.